src/main/java/com/_up/megastore/services/implementations/ProductService.java

```diff
@@ -44,15 +50,29 @@ public ProductResponse saveProduct(CreateProductRequest createProductRequest, Mu
     }
 
     @Override
-    public Product findProductByIdOrThrowException(UUID productId) {
-        return productRepository.findById(productId)
-                .orElseThrow(() -> new NoSuchElementException("Product with id " + productId + " does not exist."));
+    public ProductResponse getProduct(UUID productId) {
+        Product product = findProductByIdOrThrowException(productId);
+        return ProductMapper.toProductResponse(product);
+    }
+
+    @Override
+    public Page<ProductResponse> getProductByPages(int page, int pageSize, String sortBy, String filter) {
+        Sort sort = Sort.by(sortBy);
+        Pageable pageable = PageRequest.of(page, pageSize, sort);
+        String formattedFilter = "%" + filter + "%";
+        return productRepository.findProductsByDeletedIsFalseAndNameLikeIgnoreCase(formattedFilter, pageable)
+                .map(ProductMapper::toProductResponse);
```

Comment:
Entiendo que la concatenacion de los % es para buscar coincidencias por el nombre... Esto es porque estás usando el Like en el repositorio, probá el containing

```suggestion
        return productRepository.findProductsByDeletedIsFalseAndNameContainingIgnoreCase(filter, pageable)
                .map(ProductMapper::toProductResponse);
```